### PR TITLE
Refctor Solver.java to extend AbstractPointer

### PR DESCRIPTION
**Pull request by mudathirmahgoub**
This PR refactor Solver class in the java api to extend AbstractPointer similar to other cvc5 classes. 
It also cleans up redundant code for Abstract pointers. and adds Context.deletePointers to java examples as mentioned in issue https://github.com/cvc5/cvc5/issues/10052

### CODE REVIEW

File: examples/api/java/FloatingPointArith.java [hunk 1]
```diff
@@ -20,6 +20,8 @@
 
 import static io.github.cvc5.Kind.*;
 
+import java.math.RoundingMode;
+
```
Comment (Selebrator) [inline examples/api/java/FloatingPointArith.java:24]:
```suggestion
```

This should be `io.github.cvc5.RoundingMode`, which is already covered by the star import below.

**Comment by ajreynol** [comment]
Why is the example being deleted? Was this spurious?

**Comment by mudathirmahgoub** [comment]
> Why is the example `private static Solver getSolver()` being deleted? Was this spurious?

The method is not used anywhere.

**Review by ajreynol — APPROVED** [review]

**Comment by Selebrator** [comment]
While this looks like an improvement to me :+1:, this is what I find myself doing right now to prevent the leaks.

```java
Solver solver = new Solver();
// try-finally because we don't want to leak memory when an exception happens.
try {
  // use solver
} finally {
  // TODO Context.abstractPointers is a global variable.
  //  That means we can not have multiple solvers finishing at different times.
  //  This would be a problem if we did multithreading.
  // full path to make obvious what context.
  io.github.cvc5.Context.deletePointers();
  // Solver is the only implementation of IPointer that is not an AbstractPointer,
  // so we need to clear it manually.
  solver.deletePointer();
}
```

Not including `solver.deletePointer()` makes a big difference, as this appears to be the main resource hog. In one of my examples missing that line leaks 350 MB over 10 seconds.

What do you think about refactoring such that `solver.deletePointer()` is included in `Context.deletePointers()`?

Wrapping in a `try-finally` block would still be preferable imo.

**Comment by mudathirmahgoub** [comment]
> What do you think about refactoring such that `solver.deletePointer()` is included in `Context.deletePointers()`?

I agree. Thanks for pointing this out. Solver should be deleted like other cvc5 classes.

**Comment by Selebrator** [comment]
Looks good. I compiled and ran with it as my dependency. Works as intended. :+1:

**Review by aniemetz — APPROVED** [review]

**Pull request merged**